docs/argocd.md

```diff
@@ -0,0 +1,207 @@
+# Integrating GitOps Flow with the SiteConfig Operator
+
+This guide provides instructions for integrating a GitOps workflow with the SiteConfig operator.
+We recommend deploying the ArgoCD operator using the
+[Red Hat OpenShift GitOps operator](https://catalog.redhat.com/software/operators/detail/5fb288c70a12d20cbecc6056).
+
+## Table of Contents
+1. [Prepare Git Repository](#prepare-git-repository)
+2. [Configure ArgoCD](#configuring-argocd)
+    - [Create ArgoCD Project](#create-argocd-project)
+    - [Grant ArgoCD Permissions](#grant-argocd-permissions)
+    - [Configure Access to the Git Repository](#configure-access-to-the-git-repository)
+    - [Create ArgoCD Application](#create-argocd-application)
+
+## Prepare Git Repository
+
+To start, create a Git repository with a directory structure similar to the suggested example shown below.
+
+```sh
+clusters/
+ |- spoke1/
+ |-- extra-manifests-cm1.yaml
+ |-- extra-manifests-cm2.yaml
+ |-- clusterinstance.yaml
+ |-- ns.yaml
+ |-- secrets.yaml
+ |-- kustomization.yaml
+ |- kustomization.yaml
+```
+
+where:
+- `ns.yaml` is the Namespace manifest
+- `secrets.yaml` consists of the BMC credential and pull secrets.
+
+### Example kustomization.yaml files
+
+Below are examples of `kustomization.yaml` files required for the setup:
+
+#### clusters/kustomization.yaml
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - spoke1/
+```
+
+#### clusters/spoke1/kustomization.yaml
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ns.yaml
+  - secrets.yaml
+  - extra-manifests-cm1.yaml
+  - extra-manifests-cm2.yaml
+  - clusterinstance.yaml
+```
+
+### Generating extra-manifests ConfigMap using kustomize
```

Comment:
This section is an "alternative" or "advanced" option. Having it inline with the initial description of the flow makes it harder for the reader to follow the basic steps before they get into the "other options". I'd recommend having a section at the end for "advanced options" which includes this content.

Reply:
Moved to the end under a new Section `Advanced Topics`